From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matteo Melli <matteom@ongres.com>
Date: Tue, 21 Apr 2026 00:00:00 +0000
Subject: [PATCH 8/8] fix: remove broken gb18030_2022 bitcode custom_target

contrib/gb18030_2022/meson.build contains a copy-paste bug from
contrib/ivorysql_ora/meson.build: a custom_target that runs
"perl ../contrib/ivorysql_ora/gensql.pl" with output
"utf8_and_gb18030_2022.bc". The perl path is relative to the current
source directory, so it only resolves correctly from ivorysql_ora/; from
gb18030_2022/ it expands to <srcroot>/contrib/contrib/ivorysql_ora/...
and the build fails with:

  Can't open perl script "../contrib/ivorysql_ora/gensql.pl":
  No such file or directory

Neither the custom_target nor the accompanying run_target blocks are
needed here: shared_module() already drives the LLVM bitcode emission
when the llvm feature is enabled (see the Makefile, which only wires
the .bc file aliases for JIT). Drop both blocks so the module builds.
---
 contrib/gb18030_2022/meson.build | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/contrib/gb18030_2022/meson.build b/contrib/gb18030_2022/meson.build
--- a/contrib/gb18030_2022/meson.build
+++ b/contrib/gb18030_2022/meson.build
@@ -10,21 +10,6 @@ if host_system == 'windows'
     '--FILEDESC', 'gb18030_2022 - support gb18030 2022 with extension',])
 endif
 
-custom_target('gb18030_2022.bc',
-  output: 'utf8_and_gb18030_2022.bc',
-  command: [perl, '../contrib/ivorysql_ora/gensql.pl', 'meson', '1.0'],
-  capture: true,
-  install: true,
-  install_dir: contrib_data_args['install_dir'],
-)
-
-foreach file : ['utf8_and_gb18030_2022.bc', 'gb18030_2022.bc']
-  run_target(
-    file,
-    command: ['touch', meson.current_build_dir() / file]
-  )
-endforeach
-
 gb18030_2022 = shared_module('gb18030_2022',
   gb18030_2022_src_files,
   kwargs: contrib_mod_args,
-- 
2.39.5

